sky/packages/sky/lib/src/rendering/custom_layout.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'box.dart';
+import 'object.dart';
+
+class _MultiChildParentData extends ContainerBoxParentDataMixin<RenderBox> { }
+
+abstract class MultiChildLayoutDelegate {
+  Map<Object, RenderBox> _idToChildMap = new Map<Object, RenderBox>();
+  RenderBox _idToChild(Object childId) { return  _idToChildMap[childId]; }
```

Comment:
I'd intended to add an assert to _idToChild() since the rest of the code doesn't expect null. However allowing optional children seems like a good idea.

I'd originally wanted to allow developers to specify children by map key or index, I'd anticipated _idToChild() supporting both. Perhaps like this:

```
  RenderBox _idToChild(Object childId) {
    return _idToChildMap[childId] ?? (childId is int ? TBDchildren[childId] : null);
  }
```